File: bench/ports/java/Lists.java
import java.util.ArrayList;
import java.util.List;
import java.util.stream.Collectors;

public class Lists {
    public static void main(String[] args) {
        List<Long> xs = new ArrayList<>();
        for (long i = 0; i < 1000000L; i++) xs.add(i % 1000);

        List<Long> doubled = xs.stream().map(it -> it * 2).collect(Collectors.toList());
        List<Long> big = doubled.stream().filter(it -> it > 1000).collect(Collectors.toList());

        long acc = 0;
        for (long n : big) acc = acc + n;
        System.out.println(acc);
    }
}

File: bench/ports/python/lists.py
xs = []
for i in range(1000000): xs.append(i % 1000)
doubled = list(map(lambda it: it * 2, xs))
big = list(filter(lambda it: it > 1000, doubled))
from functools import reduce
print(reduce(lambda acc, n: acc + n, big, 0))
